### PR TITLE
Enhance Logging

### PR DESCRIPTION
**Pull request by FabienM**
Use logrus.FieldLogger interface
Contextual logging inside transaction
Gin recovery handler to logrus
Refactor a bit binary media binding

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-go-api#352.0 ✘  
* Build sources
  * Build API ✘

**Comment by ovh-cds** [comment]
CDS Report build-image-preview#353.0 ✘  
* Build ui sources
  * Build Web UI ✔ 
* Build docker
  * Build Docker Image ✘

**Comment by ovh-cds** [comment]
CDS Report build-image-preview#353.1 ✘  
* Build ui sources
  * Build Web UI ✔ 
* Build docker
  * Build Docker Image ✘

File: api/db/transactions.go [hunk 1]
```diff
@@ -28,24 +27,25 @@ func (tm *transactionManager) DB() *gorm.DB {
...
-		tm.log.Warnf("transaction has been rollbacked")
+		log.Warnf("transaction has been rollbacked")
 		return err
 	}
-	tm.log.Debugf("committing transaction")
+	log.Debug("committing transaction")
 	err := tx.Commit().Error
-	if err != nil {
-		tm.log.Debugf("transaction has ben committed")
+	if err == nil {
```
Comment (rayene) [inline api/db/transactions.go]: We were emitting erroneous logs for transactions ?

Comment (FabienM) [reply api/db/transactions.go]: nop, but the debug log on transaction commit success was happening only on commit failure 😁

**Pull request merged**